Documentation/pinning.md

```diff
@@ -0,0 +1,59 @@
+# Object pinning
+
+BPF has a persistent view of maps and programs under its own filesystem
+`/sys/fs/bpf`. Users are able to make each object visible under the bpffs.
+We call it `object pinning`. This is done by calling syscall `bpf(2)` with
+a command `BPF_OBJ_PIN`. After doing that, users are able to use the object
+with commands such as `BPF_OBJ_GET`, or remove the object with an ordinary
+VFS syscall `unlink(2)`.
+
+Doing that, we can make maps and programs stay alive across process
+terminations. This mechanism provides a much more consistent way of sharing
+object with other processes, compared to other solutions such as `tc`,
+where objects are shared via Unix domain sockets.
+
+## Different pinning options
+
+`C.bpf_map_def.pinning` can be set to one the following pinning options.
+
+* `PIN_NONE` : object is not pinned
+* `PIN_OBJECT_NS` : pinning that is local to an object (to-be-implemented)
+* `PIN_GLOBAL_NS` : pinning with a global namespace under e.g. `/sys/fs/bpf/ns1/globals`
+* `PIN_CUSTOM_NS` : pinning with a custom path given as section parameter
+
+### pinning with `PIN_CUSTOM_NS`
+
+When loading a module with `C.bpf_map_def.pinning` set to `PIN_CUSTOM_NS`,
+an additional path must be set to `elf.SectionParams.PinPath` as parameter
+to `Load()`. For example:
```

Comment:
I would also give an example of the map definition in the `.c` file

Reply:
@alban Done. I added a simple map definition to the example.

Reply:
:+1: